### ISSUE TITLE
How can I abort (reject) a request? 

### ISSUE THREAD

**Issue by VladimirShestakov** [mzabriskie/axios#193]
How can I abort (reject) a request in the outer function? 

for example:
```
var r = axios.get('url').then(()=>{...});
r.reject();
```

**Comment by mzabriskie** [comment]
You can do something like this:

```js
axios.get('/example').then((res) => {
  return Promise.reject();
}).catch(function (err) {
  console.log(err);
})
```

**Comment by gdamjan** [comment]
@mzabriskie that still waits for the result.
the point of rejecting/aborting a request is to destroy a potentionaly stuck request (cause of stuck TCP sessions, for ex.)

XHR supports .abort for ex. 
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest#abort%28%29

**Comment by mzabriskie** [comment]
Sorry I misunderstood your question. Aborting a request is challenging with Promises. Working on getting is solved for a 1.0.0 release. See existing issue for more details on this https://github.com/mzabriskie/axios/issues/50
